modules/ROOT/pages/authentication-authorization/manage-users.adoc

```diff
@@ -971,65 +980,66 @@ The `RENAME USER` command is only available when using native authentication and
+The `SET` and `REMOVE` clauses can be applied in any order.
+However, all `REMOVE` clauses must come before the first `SET` clause and at least one `SET` or `REMOVE` clause is required for the command.
 
 [source, syntax, role="noheader"]
 ----
-ALTER USER name [IF EXISTS]
-  [REMOVE HOME DATABASE]
-  [REMOVE { AUTH [PROVIDER[S]] provider[, ...] \| ALL AUTH [PROVIDER[S]] }]...
-  [SET [PLAINTEXT | ENCRYPTED] PASSWORD 'password']
-  [[SET PASSWORD] CHANGE [NOT] REQUIRED]
-  [SET STATUS {ACTIVE | SUSPENDED}]
-  [SET HOME DATABASE name]
-  [SET AUTH [PROVIDER] 'provider' "{"{SET <key> <value>}..."}"]...
-----
-`<key><value>` pairs for the `SET AUTH` clause could include:
+ALTER USER name [IF EXISTS] # <1>
+  [REMOVE HOME DATABASE] # <2>
+  [REMOVE { AUTH [PROVIDER[S]] provider[, ...] \| ALL AUTH [PROVIDER[S]] }]... # <3>
+  [SET [PLAINTEXT | ENCRYPTED] PASSWORD 'password'] # <4>
+  [[SET PASSWORD] CHANGE [NOT] REQUIRED] # <5>
+  [SET STATUS {ACTIVE | SUSPENDED}] # <6>
+  [SET HOME DATABASE name] # <7>
+  [SET AUTH [PROVIDER] 'provider' "{"{SET <key> <value>}..."}"]... # <8>
+----
+Where:
+
+<1> Specifies the command to alter a user.
+<2> Removes the home database for the user.
+As a result, the DBMS default database will be used as the home database for that user.
+<3> label:new[Introduced in 5.24] Removes one, several, or all existing xref:authentication-authorization/auth-providers.adoc[auth provider(s)] from a user.
+However, a user must always have at least one auth provider.
+Therefore, `REMOVE ALL AUTH` must be used in conjunction with at least one `SET AUTH` clause in order to meet this requirement.
+<4> Specifies the password for the user.
+The `'password'` can either be a string value or a string parameter with default value length of at least 8 characters. +
+The `PLAINTEXT` and `ENCRYPTED` keywords are optional and can be used to specify the format of the password, i.e. whether Neo4j needs to hash it or it has already been hashed.
+By default, all passwords are encrypted (hashed) when stored in the Neo4j `system` database.
+* The optional `PLAINTEXT` in `SET PLAINTEXT PASSWORD` has the same behavior as `SET PASSWORD`.
+* The optional `ENCRYPTED` is used to recreate an existing user when the plaintext password is unknown, but the encrypted password is available in the _data/scripts/databasename/restore_metadata.cypher_ file of a database backup.
+See xref:backup-restore/restore-backup#_restore_users_and_roles_metadata[Restore users and roles metadata]. +
+With `ENCRYPTED`, the password string is expected to be in the format of `<encryption-version>,<hash>,<salt>`, where, for example:
+** `0` is the first version and refers to the `SHA-256` cryptographic hash function with iterations `1`.
+** `1` is the second version and refers to the `SHA-256` cryptographic hash function with iterations `1024`.
+<5> Specifies whether the user must change their password at the next login.
+If omitted, the default is `CHANGE REQUIRED`.
```

Comment:
for ALTER this is actually more nuanced.

* If the user has native auth already (previously always true), and no change require clause is given then the value of the flag isn't updated and the old value is kept.
* if the user doesn't have native auth already (so only has external auths) **and** a set password clause is given, then we default to `CHANGE REQUIRED` when this clause is omitted.
* if the user doesn't have native auth already (so only has external auths), and we have neither a set password nor a change required clause then it behaves as the create case and doesn't add the change required (defaults to null)
* lastly same as for create, if the user doesn't have native auth already (so only has external auths) and only the change required is given this is an error as we don't have a password

Reply:
But these 2 sentences just want to explain what the syntax means with the native authentication as with create.

Reply:
yeah, but it is wrong (and I think it was before as well). In the alter case we don't have add a default unless you add native auth to the user where it didn't have it before, otherwise we just keep what the user already have for this when omitted